addOns/selenium/src/main/java/org/zaproxy/zap/extension/selenium/SeleniumAPI.java

```diff
@@ -38,6 +47,27 @@ public SeleniumAPI() {
      */
     public SeleniumAPI(SeleniumOptions options) {
         addApiOptions(options);
+        parser = new StandardParameterParser();
+    }
+
+    @Override
+    public String handleCallBack(HttpMessage msg) throws ApiException {
+        try {
+            int id;
+            List<NameValuePair> params = parser.getParameters(msg, Type.url);
+            for (NameValuePair pair : params) {
+                if (pair.getName().equalsIgnoreCase("hist")) {
+                    id = Integer.parseInt(pair.getValue());
+                    HttpMessage req = new HistoryReference(id, true).getHttpMessage();
+                    msg.setResponseBody(req.getResponseBody());
+                    msg.setResponseHeader(req.getResponseHeader());
+                    return req.getResponseBody().toString();
+                }
+            }
+            return null;
+        } catch (Exception e) {
+            throw new ApiException(ApiException.Type.URL_NOT_FOUND, e);
```

Comment:
I ended up with a ton of `1286162 [ZAP-IO-Server-1-17] WARN  org.zaproxy.zap.extension.api.API - Request to callback URL https://demo.testfire.net/zapCallBackUrl/images/home2.jpg from 127.0.0.1 not found - this could be a callback url from a previous session or possibly an attempt to attack ZAP` while testing. I guess this is due to media included in the custom request I was opening? It's awfully chatty/loggy.

Reply:
I didn't understand how you got this :)..... the hist param didn't get appended :).....